internal/payloads/reservation_payload.go

```diff
@@ -2,33 +2,58 @@ package payloads
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/go-chi/render"
 )
 
 // ReservationRequest is empty, account comes in HTTP header and
 // provider type in HTTP URL. All other fields are auto-generated.
-type ReservationRequest struct {
-}
 
-type ReservationResponse struct {
-	*models.Reservation
-}
+type ReservationPayload struct {
+	ID int64 `json:"id"`
 
-type NoopReservationRequest ReservationRequest
+	// Provider type. Required.
+	Provider int `json:"provider"`
 
-type NoopReservationResponse struct {
-	*models.NoopReservation
+	// Account ID. Required.
+	AccountID int64 `json:"-"`
+
+	// Time when reservation was made.
+	CreatedAt time.Time `json:"created_at"`
+
+	// Textual status of the reservation or error when there was a failure
+	Status string `json:"status"`
+
+	// Time when reservation was finished or nil when it's still processing.
+	FinishedAt time.Time `json:"finished_at"`
+
+	// Flag indicating success, error or unknown state (NULL). See Status for the actual error.
+	Success bool `json:"success"`
 }
```

Comment:
All the fields which you want to be NULLable in JSON must be pointers. Only then they will be marshalled as JSON "null" values (when "nil" in Go).